src/Microsoft.AspNetCore.Authentication/Events/HandleRequestContext.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public class HandleRequestContext<TOptions> : BaseContext<TOptions> where TOptions : AuthenticationSchemeOptions
+    {
+        protected HandleRequestContext(
+            HttpContext context,
+            AuthenticationScheme scheme,
+            TOptions options)
+            : base(context, scheme, options) { }
+
+        /// <summary>
+        /// The <see cref="HandleRequestResult"/> which is used by the handler.
+        /// </summary>
+        public HandleRequestResult Result { get; protected set; }
+
+        /// <summary>
+        /// Discontinue all processing for this request and return to the client.
+        /// The caller is responsible for generating the full response.
+        /// </summary>
+        public void HandleResponse() => Result = HandleRequestResult.Handle();
+
+        /// <summary>
+        /// Discontinue processing the request in the current handler.
+        /// </summary>
+        public void SkipRequest() => Result = HandleRequestResult.SkipRequest();
```

Comment:
`SkipToNextMiddleware()` still makes sense because in most cases, folks who use this API want to invoke the other middleware (e.g to render an error page or run some logic in a MVC controller), not an other `IAuthenticationRequestHandler`.

If you really want to introduce a breaking change in user code, `SkipToNextMiddleware()` could renamed to ``SkipToNextComponent()`.

Or you can rename `IAuthenticationRequestHandler` to something that has the middleware in the name and avoid the breaking change.